src/compiler/utilities.ts

```diff
@@ -84,6 +84,25 @@ namespace ts {
         return node.end - node.pos;
     }
 
+    export function mapIsEqualTo<T>(map1: Map<T>, map2: Map<T>): boolean {
+        if (!map1 || !map2) {
+            return map1 === map2;
+        }
+        return containsAll(map1, map2) && containsAll(map2, map1);

```

Comment:
would it be faster to do something like `object.getownpropertynames(map1).length === object.getownpropertynames(map2).length) && containsAll(map1, map2))